scripts/pdf_utils.py

```
@@ -420,28 +455,34 @@ def compress_pdf(input_path: str, output_dir: str, level: str = 'medium', passwo
-def pdf_to_docx(input_path: str, output_dir: str, password: str = None) -> str:
-    """Converts PDF to DOCX using pdf2docx (Fast, Rule-based)."""
+def _convert_pdf2docx(decrypted_path: str, output_file: Path) -> None:
+    """Shared pdf2docx conversion core, used by both the standard converter
+    and pdf_to_word_ai's automatic text-layer routing."""
     from pdf2docx import Converter
 
+    cv = Converter(decrypted_path)
+    try:
+        try:
+            cv.convert(str(output_file), multi_processing=True)
+        except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)